Explorer/Assets/Plugins/CommandTerminal/ConsoleBuffer.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+// ORIGINAL OPEN SOURCE PLUGIN SCRIPT:
+// https://github.com/stillwwater/command_terminal/blob/0f5918ea79014955b24c7d431625a97a1cac8797/CommandTerminal/CommandLog.cs
+
+namespace CommandTerminal
+{
+    public struct LogItem
+    {
+        public LogType type;
+        public string message;
+        public string stackTrace;
+    }
+
+    public class ConsoleBuffer
+    {
+        private readonly int maxItems;
+
+        public List<LogItem> Logs { get; } = new ();
```

Comment:
Uhmm, I'm not sure about that change, that would force me to cast the `Logs` collection into `List<LogItem>` every time I need to `Add`, `RemoveAt()` or `Clear()` on it, like this:
```
public IReadOnlyList<LogItem> Logs { get; } = new List<LogItem>();

public void HandleLog(string message, string stackTrace, LogType logType)
{
    List<LogItem> logs = (List<LogItem>)Logs;
    logs.Add(new LogItem(logType, message, stackTrace));
    if (logs.Count > maxItems) { logs.RemoveAt(0); }
}

public void Clear()
{
     ((List<LogItem>)Logs).Clear();
 }
```

(otherwise I'd have to implement some painful logic to add and remove the items based on indices, like an array)

Reply:
Extra cast is a bad practice here from my perspective

I meant it should be like:
`
private readonly List<LogItem> logs = new List<LogItem>();

public IReadOnlyList<LogItem> Logs => logs;

public void HandleLog(string message, string stackTrace, LogType logType)
{
    logs.Add(new LogItem(logType, message, stackTrace));
    if (logs.Count > maxItems) { logs.RemoveAt(0); }
}

public void Clear()
{
    logs.Clear();
}
`

Reply:
The point is the class exposes log's readonly state and cannot be corrupted from outside

Otherwise it's possible to ignore `HandleLog` and add an item directly to the list without the extra logic with `maxItems`

Reply:
Ahh I see what you mean, yeah good point.
Implemented at: https://github.com/decentraland/unity-explorer/pull/1546/commits/7db392aa7c131317dd50483f83fe60edbcb8d882